internal/test/e2e/e2e_test.go

```diff
@@ -0,0 +1,226 @@
+package e2e
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	etcd "go.etcd.io/etcd/client"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kindv1alpha3 "sigs.k8s.io/kind/pkg/apis/config/v1alpha3"
+	"sigs.k8s.io/kind/pkg/cluster"
+	"sigs.k8s.io/kind/pkg/cluster/create"
+	"sigs.k8s.io/kind/pkg/container/cri"
+
+	"github.com/improbable-eng/etcd-cluster-operator/internal/test/try"
+)
+
+const (
+	expectedClusterSize = 3
+)
+
+var (
+	fUseKind           = flag.Bool("kind", false, "Creates a Kind cluster to run the tests against")
+	fUseCurrentContext = flag.Bool("current-context", false, "Runs the tests against the current kubernetes context, the path to kube config defaults to ~/.kube/config, unless overridden by the KUBECONFIG environment variable")
+	fRepoRoot          = flag.String("repo-root", "", "The absolute path to the root of the etcd-cluster-operator git repository.")
+	fCleanup           = flag.Bool("cleanup", true, "Tears down the Kind cluster once the test is finished.")
+
+	etcdConfig = etcd.Config{
+		Endpoints: []string{"http://127.0.0.1:2379"},
+		Transport: etcd.DefaultTransport,
+		// set timeout per request to fail fast when the target endpoint is unavailable
+		HeaderTimeoutPerRequest: time.Second,
+	}
+)
+
+func TestE2E_Kind(t *testing.T) {
+	if !*fUseKind {
+		t.Skip()
+	}
+
+	// Tag for running this test, for naming resources.
+	operatorImage := "etcd-cluster-operator:test"
+
+	// Create Kind cluster to run the workloads.
+	kind, stopKind := setupLocalCluster(t)
+	defer stopKind()
```

Comment:
If I hit ctrl-c, the kind cluster doesn't get cleaned up.

Reply:
added a SIGINT and SIGTERM handler to make sure this gets called